tribits/common_tpls/FindTPLHDF5.cmake

```diff
@@ -16,10 +18,13 @@ if (Netcdf_ALLOW_MODERN)
       HDF5
       INNER_FIND_PACKAGE_NAME  HDF5
       IMPORTED_TARGETS_FOR_ALL_LIBS   ${HDF5_EXPORT_LIBRARIES})
+    set(INTERNAL_IS_MODERN TRUE)
   endif()
 
 endif()
 
+set(HDF5_FOUND_MODERN_CONFIG_FILE ${INTERNAL_IS_MODERN} CACHE BOOL "True if HDF5 was found by the modern method" FORCE)
```

Comment:
`${INTERNAL_IS_MODERN}` should be quoted to ensure that `set()` receives a value... though I see from above that it will always have a value, so perhaps it doesn't matter. Still, quoting it makes this explicit.

Reply:
This is a good point I never considered. If `INTERNAL_IS_MODERN` is empty (or undefined), this statement just produces a regular list variable with the value:

```
"CACHE;BOOL;True if HDF5 was found by the modern method;FORCE"
```

Is that correct?

Reply:
> So it turns out that if it's empty or undefined, it will simply set a cache variable with an empty value. However, readers won't know this just from glancing at the code, and for that reason I still recommend quoting it.

Then what is the benefit of quoting the `${val}` argument in `set(<var> ${val} CACHE ...)` if it does not impact behavior?  In my example [above](https://github.com/TriBITSPub/TriBITS/pull/535#discussion_r1007042793), quoting the argument can  change behavior.

Reply:
> then what is the benefit of quoting the ${val} argument in set(<var> ${val} CACHE ...) if it does not impact behavior?

It assures the reader that there won't be a problem. I had to test the behavior of `set(... CACHE)` to check that it would work without quoting, whereas I knew instantly just by looking at it that it would work *with* quoting. I believe it's better to be explicit than implicit. Just a thought.

Reply:
That is a good point.  There really needs to be a CMake linter to catch issues like this automatically.  We should not expend too much time in PR reviews dealing with low-level language issues like this.  That is why they make linters in the first place.

In the future, we will need your help reviewing these types of changes when they are being created and tested in the Trilinos github repo.  Once we being this back to TriBITS (as the official repo for these common TPLs) we can't test them easily.